### DISCUSSION TITLE
Bootstrap Button with 'nav-link' and 'btn-outline' Classes Not Displaying Borders

### DISCUSSION THREAD

**Discussion by alfon96** [twbs/bootstrap#39377] [Q&A]
## Environment
- Bootstrap version: 5.3.0
- Browser and version: Edge 119.0.2151.44

## Description
I am using Bootstrap for navigation links in my project, and I've encountered an issue with a button styled as a link. When I apply the classes `nav-link` and `btn btn-outline-light` together, the button's borders are not visible, which is not the expected behavior. If I remove the `nav-link` class, the borders become visible, but then the text alignment does not match other navigation links.

Here's the code snippet:

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <link
      href="https://fonts.googleapis.com/css2?family=Open+Sans:ital,wght@0,400;0,600;0,700;1,400&display=swap"
      rel="stylesheet"
    />
    <link rel="stylesheet" href="css/font-awesome.css" />
    <link rel="stylesheet" href="css/bootstrap.css" />
    <link rel="stylesheet" href="css/styles.css" />
    <link rel="icon" href="images/favicon.png" />
    <title>Free E-Book | Start Your Own Blog</title>
  </head>
  <body>
    <!-- Navigation -->
    <nav class="navbar navbar-expand-lg fixed-top navbar-dark bg-dark">
      <div class="container">
        <a href="index.html" class="navbar-brand">
          <img src="images/logo.png" alt="" width="225" />
        </a>
        <button
          class="navbar-toggler"
          type="button"
          data-bs-toggle="collapse"
          data-bs-target="#navbarNavDropdown"
        >
          <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="navbarNavDropdown">
          <ul class="navbar-nav ms-auto">
            <li class="nav-item">
              <a href="index.html" class="nav-link fw-semibold">Home</a>
            </li>
            <li class="nav-item">
              <a href="#details" class="nav-link fw-semibold">Details</a>
            </li>
            <li class="nav-item">
              <a
                href="contact.html"
                class="nav-link btn btn-outline-light mx-4 px-4 fw-semibold"
                >Contact</a
              >
            </li>
          </ul>
        </div>
      </div>
    </nav>

    <script src="js/bootstrap.bundle.min.js"></script>
    <script src="js/script.js"></script>
  </body>
</html>

```

## Expected Behavior
I expect the button to display borders from the btn-outline-light class while also aligning text as per the nav-link class.

## Actual Behavior
Borders are not visible when both nav-link and btn btn-outline-light classes are applied together.

## Attempts to Resolve
Checked for any conflicting CSS rules.
Ensured that the Bootstrap version is the latest and properly linked.
Searched through Bootstrap's documentation for known issues.

# Question
Is there an inherent conflict when using nav-link with btn and btn-outline classes, or am I potentially missing something? I would prefer not to use a workaround like applying additional flexbox styling to solve this issue but rather understand if there's a mistake in my implementation.

Any insights or suggestions would be greatly appreciated!

**Comment by julien-deramond** [comment]
Hey @alfon96 
IMO you should not use `nav-link btn btn-outline-light` but `btn btn-outline-light` (like the third example of https://getbootstrap.com/docs/5.3/examples/headers/ but with buttons next to the nav links).

> text alignment does not match other navigation links

Which alignment are you talking about? Vertical positioning of the text baselines? Or something else?

**Reply by alfon96** [reply]
Thanks for the advice on using Bootstrap classes.

 I encountered an issue after following a Udemy course that used Bootstrap 5.2.3 where nav-link combined with other classes worked well. However, in Bootstrap 5.3.0, this approach didn't display outline borders correctly.

To illustrate:

- Bootstrap 5.2.3 using nav-link, everything works fine (https://github.com/alfon96/Blog-Mastery-Older-Bootstrap).
- Bootstrap 5.3.0 using nav-link, the border outline is missing (https://github.com/alfon96/Blog-Mastery-Website-Issue).
- Bootstrap 5.3.0 **not** using nav-link, functionality is okay but vertical alignment is off (see attached picture)  (https://github.com/alfon96/Blog-Mastery-Website).
(Please run npm install and npm run sass:build before reviewing the index.html files.)


![btn-outline-light issue](https://github.com/twbs/bootstrap/assets/119021026/96a548d0-c703-4591-ad0c-a46ab479828d)

The alignment issue is visible in the attached image. While 'Home' and 'Details' align well, the 'Contact' button text is misaligned.

I appreciate your help in resolving this.

**Reply by julien-deramond** [reply]
I haven't checked too much in detail but I think overriding the buttons' padding-x values can be messing up a little bit with the alignments here. There's maybe a smarter/more elegant solution, but for a quick fix, I think you should be able to do something like this to force the vertical centering only at large breakpoint:

```scss
@include media-breakpoint-up(lg) { /* or media queries instead if you don't have access to this from Bootstrap */
  .navbar-nav { / * prefer using a more specific class or id instead of this selector */
    align-items: center;
  }
}
```
